lib/api/CFieldDataCategorizer.cc

```diff
@@ -347,8 +461,39 @@ bool CFieldDataCategorizer::persistState(core::CDataAdder& persister,
 
     LOG_DEBUG(<< "Persist categorizer state");
 
-    return this->doPersistState(m_DataCategorizer->makeForegroundPersistFunc(),
-                                m_DataCategorizer->examplesCollector(), persister);
+    TStrVec partitionFieldValues;
+    TPersistFuncVec dataCategorizerPersistFuncs;
+    TCategoryExamplesCollectorsCRefVec examplesCollectors;
+
+    if (m_PartitionFieldName.empty()) {
+        model::CDataCategorizer& dataCategorizer{categorizerForKey(EMPTY_STRING)};
+        dataCategorizerPersistFuncs.emplace_back(dataCategorizer.makeBackgroundPersistFunc());
```

Comment:
> Isn't this a foreground version?

Yes, good spot.  I guess it didn't cause any problems because background works too, just with more copying 🤦 

> I think it would be good to call this something like `persistStateInForeground`

👍